executor/runtime/types/pod.go

```diff
@@ -0,0 +1,330 @@
+package types
+
+import (
+	"encoding/base64"
+	"regexp"
+	"time"
+
+	"github.com/Netflix/titus-executor/api/netflix/titus"
+	"github.com/Netflix/titus-executor/uploader"
+	vpcTypes "github.com/Netflix/titus-executor/vpc/types"
+	"github.com/golang/protobuf/proto" // nolint: staticcheck
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// PodContainer is an implementation of Container backed only by a kubernetes pod.
+// This is currently using the base64'ed ContainerInfo until all fields are ported over to annotations
+type PodContainer struct {
+	command       []string
+	entrypoint    []string
+	hostnameStyle string
+	ipv4Address   *string
+	pod           *corev1.Pod
+	titusInfo     *titus.ContainerInfo
+}
+
+func NewPodContainer(pod *corev1.Pod, ipv4Address *string) (*PodContainer, error) {
+	if pod == nil {
```

Comment:
If pod is nil, that seems..bad, maybe panic?

Reply:
In the case of the executor, shouldn't we return a nice-ish message?